### PR TITLE
fix the addon is  failed to deploy on OCP311

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>
issue:  https://github.com/stolostron/backlog/issues/21554

### CODE REVIEW

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -69,8 +69,9 @@ func NewGetValuesFunc(imageName string, imageRegistryClient imageregistry.Client
 		}
 
 		for _, claim := range cluster.Status.ClusterClaims {
-			if claim.Name == "product.open-cluster-management.io" && claim.Value == "OpenShift" {
-				addonValues.IsOCP = true
+			if claim.Name == "version.openshift.io" && claim.Value != "3" {
```
Comment (qiujian16) [inline pkg/addon/addon.go:72]: this is a danger claim, we cannot say it is OCP4 if it is not 3.

Comment (zhiweiyin318) [reply pkg/addon/addon.go:72]: the cluster is OCP if there is version.openshift.io clusterclaim.  only 2 options for the version, 3 or 4.x.x currently.

Comment (qiujian16) [reply pkg/addon/addon.go:72]: I would use HasRoute as flag instead using ISOCP4. Also add a comment in this section on how we detect whether route exists

Comment (zhiweiyin318) [reply pkg/addon/addon.go:72]: done

**Pull request merged**